### PR TITLE
[Feature][CDC] Support for preferring numeric fields as split keys

### PR DESCRIPTION
**Pull request by Carl-Zhou-CN**
## Purpose of this pull request

colse #5156 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
good

**Comment by EricJoy2048** [comment]
@ic4y

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/enumerator/splitter/AbstractJdbcSourceChunkSplitter.java [hunk 1]
```diff
@@ -356,16 +357,23 @@ protected Column getSplitColumn(
...
-                    return column;
+                    splitColumn = columnComparable(splitColumn, column);
+                    if (sqlTypePriority(splitColumn) == 1) {
+                        return splitColumn;
+                    }
                 }
             }
+            if (splitColumn != null) {
+                return splitColumn;
+            }
```
Comment (ic4y) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/enumerator/splitter/AbstractJdbcSourceChunkSplitter.java]: Remove this, would it be better to also include Unique Keys in the comparison?

Comment (Carl-Zhou-CN) [reply seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/enumerator/splitter/AbstractJdbcSourceChunkSplitter.java]: done

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-base/pom.xml [hunk 2]
```diff
@@ -30,6 +30,7 @@
 
     <properties>
         <hikaricp.version>4.0.3</hikaricp.version>
+        <junit.vserion>4.13.2</junit.vserion>
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-base/pom.xml:33]: Why not use junit 5.9.0 define in root pom.xml?

Comment (Carl-Zhou-CN) [reply seatunnel-connectors-v2/connector-cdc/connector-cdc-base/pom.xml:33]: Currently, a large number of tests are used for this version

**Review by EricJoy2048 — APPROVED** [review]

**Review by ic4y — APPROVED** [review]

**Pull request merged**